ambry-router/src/main/java/com.github.ambry.router/NonBlockingRouter.java

```diff
@@ -144,17 +146,31 @@ private OperationController getOperationController() {
       throw new IllegalArgumentException("blobId or options must not be null");
     }
     currentOperationsCount.incrementAndGet();
+    final FutureResult<GetBlobResult> futureResult = new FutureResult<>();
+    GetBlobOptionsInternal internalOptions = new GetBlobOptionsInternal(options, false, routerMetrics.ageAtGet);
+    boolean isEncrypted = false;
+    try {
+      isEncrypted = new BlobId(blobId, clusterMap).isEncrypted();
```

Comment:
1) There is a function for this `RouterUtils.getBlobIdFromString()`
2) It looks like you convert to `BlobId` early in the stack, why not pass it down? I know it will cause a few constructor changes but before now, we didn't need a string -> `BlobId` conversion so early. Now we do.

Reply:
have fixed the 1st comment. Will take the 2nd as a separate commit